.github/workflows/test.yml

```diff
@@ -13,7 +13,21 @@ on:
     - cron: '30 5 * * 1'
 
 jobs:
-  test-matrix:
+  rustfmt:
+    if: github.event_name != 'schedule'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Install Rust nightly toolchain
+        run: rustup toolchain install --no-self-update nightly --profile minimal -c rustfmt
+
+      - name: Format check
+        run: cargo +nightly fmt --all -- --check
+
+  clippy:
+    if: github.event_name != 'schedule'
     strategy:
       fail-fast: false
       matrix:
```

Comment:
I'm slightly torn about running `cargo clippy` as a matrix... It feels excessive to run it on multiple targets, and it probably is. But we do use conditional compilation in a few files:

```
./cli/tests/sandbox.rs:12:    #[cfg(target_os = "linux")]
./cli/tests/sandbox.rs:14:    #[cfg(not(target_os = "linux"))]
./cli/src/update/unix.rs:109:    let arch = if cfg!(target_arch = "x86_64") {
./cli/src/update/unix.rs:111:    } else if cfg!(target_arch = "aarch64") {
./cli/src/update/unix.rs:116:    let os = if cfg!(target_os = "linux") {
./cli/src/update/unix.rs:118:    } else if cfg!(target_os = "macos") {
./cli/src/commands/sandbox.rs:7:#[cfg(target_os = "linux")]
./cli/src/commands/sandbox.rs:9:#[cfg(target_os = "linux")]
./cli/src/commands/sandbox.rs:57:        #[cfg(target_os = "linux")]
```

Does anybody feel strongly about this one way or the other?

Reply:
I think running clippy against all targets is going to be worth it when we want to be fully thorough, i.e. right before a merge or a release, otoh I find it a bit overkill for day-to-day activity as we don't really touch platform specific code very often.

Reply:
I don't feel **strongly** about either direction, however I do think it provides value to run clippy on all target operating systems.

It's also probably not worth it to bikeshed over reducing number of build steps until we're likely to run into issues. And as far as I understand it we're far below any limit of parallel GitHub runners? So my vote's to just keep it because we can and it doesn't hurt.